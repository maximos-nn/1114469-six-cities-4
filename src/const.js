const PERCENTS_PER_RATING_POINT = 20;

const PlaceType = {
  APARTMENT: `Apartment`,
  ROOM: `Private room`,
  HOUSE: `House`,
  HOTEL: `Hotel`
};

const SortType = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};

const AppRoute = {
  ROOT: `/`,
  SIGNIN: `/login`,
  OFFER: `/offer`,
  FAVORITES: `/favorites`
};

const Error = {
  UNAUTHORIZED: 401
};

export {PERCENTS_PER_RATING_POINT, PlaceType, SortType, AppRoute, Error};
