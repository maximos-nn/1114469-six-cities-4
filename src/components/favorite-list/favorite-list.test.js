import React from "react";
import renderer from "react-test-renderer";
import FavoriteList from "./favorite-list.jsx";
import {PlaceType} from "../../const.js";
import {Router} from "react-router-dom";
import history from "../../history";

const card = {
  id: 14,
  title: `Beautiful & luxurious apartment at great location`,
  type: PlaceType.APARTMENT,
  picture: `img/apartment-01.jpg`,
  price: 235,
  rating: 4.2,
  isBookmarked: true,
  isPremium: true,
  guests: 4,
  bedrooms: 3,
  description: `Description.`,
  amenities: [
    `Cabel TV`,
    `Fridge`
  ],
  host: {
    name: `Angelina`,
    avatar: `img/avatar-angelina.jpg`,
    isSuper: true
  },
  photos: [
    `img/studio-01.jpg`,
    `img/apartment-01.jpg`
  ],
  location: [52.3809553943508, 4.939309666406198],
  city: {name: `Amsterdam`}
};

it(`Should render favorite list correctly`, () => {
  const tree = renderer.create(
      <Router history={history}>
        <FavoriteList places={[card]} cities={[`Amsterdam`]} />
      </Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
