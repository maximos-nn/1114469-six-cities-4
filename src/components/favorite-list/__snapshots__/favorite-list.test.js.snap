// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render favorite list correctly 1`] = `
<ul
  className="favorites__list"
>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Amsterdam
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    >
      <article
        className="favorites__card place-card"
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="favorites__image-wrapper place-card__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="110"
              src="img/apartment-01.jpg"
              width="150"
            />
          </a>
        </div>
        <div
          className="favorites__card-info place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €235
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button place-card__bookmark-button--active button"
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                In bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/14"
              onClick={[Function]}
            >
              Beautiful & luxurious apartment at great location
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            Apartment
          </p>
        </div>
      </article>
      <article
        className="favorites__card place-card"
      />
    </div>
  </li>
</ul>
`;
